### ISSUE TITLE
Ellipsis object fails when expanded to zero axes.

### ISSUE THREAD

**Issue by jjhelmus** [Unidata/netcdf4-python#371]
The following snippet works to set netCDF variable bar to [1, 2, 3, 4, 5] using netcdf4-python version 1.1.1 but raises an IndexError in version 1.1.6:

``` Python
import netCDF4
dataset = netCDF4.Dataset('foo.nc', 'w')
dataset.createDimension('dim', 5)
var = dataset.createVariable('bar', 'i8', ('dim', ))
var[..., :] = [1, 2, 3, 4, 5]
```

Traceback is as follows with version 1.1.6:

```
Traceback (most recent call last):
  File "bug.py", line 5, in <module>
    var[..., :] = [1, 2, 3, 4, 5]
  File "netCDF4.pyx", line 3319, in netCDF4.Variable.__setitem__ (netCDF4.c:46343)
  File "/home/jhelmus/anaconda/lib/python2.7/site-packages/netCDF4_utils.py", line 179, in _StartCountStride
    dimname = dimensions[i]
IndexError: tuple index out of range
```

In NumPy, Ellipsis objects (...) are expanded to the number of **:** object to complete the selection tuple.

**Comment by jjhelmus** [comment]
bisected to 34e3e82230251ed6e570044853bb159c778118b7

**Comment by jswhit** [comment]
Good catch.  Fix in pull request #372.

**Comment by jswhit** [comment]
Merged pull request #372, closing...

**Comment by jjhelmus** [comment]
Thanks for the quick fix!
